pkgs/build-support/dotnet/build-dotnet-module/default.nix

```diff
@@ -202,131 +204,46 @@ stdenvNoCC.mkDerivation (args // {
-        src=$tmp/src
-        cp -rT "$storeSrc" "$src"
-        chmod -R +w "$src"
-
-        cd "$src"
-        echo "Restoring project..."
-
-        ${dotnet-sdk}/bin/dotnet tool restore
-        cp -r $HOME/.nuget/packages/* $tmp/nuget_pkgs || true
-
-        for rid in "${lib.concatStringsSep "\" \"" runtimeIds}"; do
-            (( ''${#projectFiles[@]} == 0 )) && dotnetRestore "" "$rid"
-
-            for project in ''${projectFiles[@]-} ''${testProjectFiles[@]-}; do
-                dotnetRestore "$project" "$rid"
-            done
-        done
-        # Second copy, makes sure packages restored by ie. paket are included
-        cp -r $HOME/.nuget/packages/* $tmp/nuget_pkgs || true
-
-        echo "Succesfully restored project"
-
-        echo "Writing lockfile..."
-
-        excluded_sources="${lib.concatStringsSep " " sdkDeps}"
-        for excluded_source in ''${excluded_sources[@]}; do
-          ls "$excluded_source" >> "$tmp/excluded_list"
-        done
-        tmpFile="$tmp"/deps.nix
-        echo -e "# This file was automatically generated by passthru.fetch-deps.\n# Please dont edit it manually, your changes might get overwritten!\n" > "$tmpFile"
-        nuget-to-nix "$tmp/nuget_pkgs" "$tmp/excluded_list" >> "$tmpFile"
-        mv "$tmpFile" "$depsFile"
-        echo "Succesfully wrote lockfile to $depsFile"
-      '';
+    fetch-deps = let
+      flagsList = dotnetFlags ++ dotnetRestoreFlags;
+      flags = lib.concatStringsSep " " (map lib.escapeShellArg flagsList);
+      disableParallel =
+        lib.optionalString (!enableParallelBuilding) "--disable-parallel";
+      runtimeIdsList = if runtimeId != null then
+        [ runtimeId ]
+      else
+        map (system: dotnetCorePackages.systemToDotnetRid system) platforms;
+      runtimeIds =
+        lib.concatStringsSep " " (map lib.escapeShellArg runtimeIdsList);
+      defaultDepsFile =
+        # Wire in the nugetDeps file such that running the script with no args
+        # runs it agains the correct deps file by default.
+        # Note that toString is necessary here as it results in the path at
+        # eval time (i.e. to the file in your local Nixpkgs checkout) rather
+        # than the Nix store path of the path after it's been imported.
+        if lib.isPath nugetDepsFile
+        && !lib.hasPrefix "${builtins.storeDir}/" (toString nugetDepsFile) then
+          toString nugetDepsFile
+        else
+          ''$(mktemp -t "${pname}-deps-XXXXXX.nix")'';
+      storeSrc = srcOnly args;
+      projectFileStr = lib.escapeShellArg projectFiles;
+      testProjectFileStr = lib.escapeShellArg testProjectFiles;
+      path = lib.makeBinPath [
+        coreutils
+        runtimeShellPackage
+        dotnet-sdk
+        (nuget-to-nix.override { inherit dotnet-sdk; })
+      ];
+      dotnetSdkPath = toString dotnet-sdk;
+      excludedSources =
+        lib.concatStringsSep " " (map lib.escapeShellArg sdkDeps);
+    in substituteAll {
```

Comment:
This seems to be resulting in the script not having the executable bit set, so `$(nix-build -A omnisharp-roslyn.fetch-deps)` fails with

> bash: /nix/store/8l6i3lrgy1ima071gvnqd25r73zsmbd2-fetchDeps.sh: Permission denied

Reply:
Oh no, sorry about that - that's pretty inexcusable for me not to have actually *tried* to execute the script, I've only invoked it with `bash` :/

Reply:
OK, I've now actually done an end-to-end test on omnisharp (NuGet restore failed because of a recent security advisory against System.Text.Json) and on my personal project.

Reply:
I tested it here too and it seems fine now.